src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -292,35 +327,178 @@ const ProxyDeviceForm: FC<Props> = ({ formik, project }) => {
               );
             }}
-            value={device.connect}
-            type="text"
-            help={
-              <ConfigFieldDescription
-                description={
-                  "Use the following format to specify the address and port: <type>:<addr>:<port>[-<port>][,<port>]"
-                }
-              />
-            }
+            value={connectType}
+            options={proxyAddressTypeOptions}
             className="u-no-margin--bottom"
+            disabled={device.nat === "true"}
+            title={
+              device.nat
+                ? "This is determined by the listen type when nat mode is enabled"
+                : undefined
+            }
           />
         ),
         override: "",
       }),
     );
+
+    customRows.push(
+      getConfigurationRowBase({
+        className: "no-border-top inherited-with-form",
+        configuration:
+          connectType === "unix" ? (
+            <Label>*Socket path</Label>
+          ) : (
+            <Label>*Address</Label>
+          ),
+        inherited:
+          connectType === "unix" ? (
+            <Input
+              onChange={(e) => {
+                ensureEditMode(formik);
+                const newAddress = e.target.value;
+                void formik.setFieldValue(
+                  `devices.${index}.connect`,
+                  `${newAddress}`,
+                );
+              }}
+              value={connectAddress}
+              placeholder="/"
+              type="text"
+              className="u-no-margin--bottom"
+            />
+          ) : (
+            <Input
+              onChange={(e) => {
+                ensureEditMode(formik);
+                const newAddress = e.target.value;
+                void formik.setFieldValue(
+                  `devices.${index}.connect`,
+                  `${connectType}:${newAddress}:${connectPort}`,
+                );
+              }}
+              value={connectAddress}
+              placeholder="000.000.000.000"
+              type="text"
+              className="u-no-margin--bottom"
+            />
+          ),
+        override: "",
+      }),
+    );
+
+    customRows.push(
+      getConfigurationRowBase({
+        className: "no-border-top inherited-with-form",
+        configuration: connectType === "unix" ? null : <Label>*Port</Label>,
+        inherited:
+          connectType === "unix" ? null : (
+            <Input
+              onChange={(e) => {
+                ensureEditMode(formik);
+                const newPort = e.target.value;
+                void formik.setFieldValue(
+                  `devices.${index}.connect`,
+                  `${connectType}:${connectAddress}:${newPort}`,
+                );
+              }}
+              value={connectPort}
+              placeholder="00[-00]"
+              type="text"
+              className="u-no-margin--bottom"
+            />
+          ),
+        override: "",
+      }),
+    );
```

Comment:
This is repeating very similar code as above.
Maybe put this into a helper file like `getProxyAddress`, which is parameterized for listen and connect, so we can call it twice and don't have to repeat all the fields and the logic around unix sockets, etc. twice.

Reply:
Working on this... :)